src/core/layout/qgscompositionconverter.cpp

```diff
@@ -0,0 +1,1414 @@
+  if ( keepLayerSetFlag.compare( QLatin1String( "true" ), Qt::CaseInsensitive ) == 0 )
+  {
+    layoutItem->setKeepLayerSet( true );
+  }
+  else
+  {
+    layoutItem->setKeepLayerSet( true );
+  }
+
+  QString drawCanvasItemsFlag = itemElem.attribute( QStringLiteral( "drawCanvasItems" ), QStringLiteral( "true" ) );
+  if ( drawCanvasItemsFlag.compare( QLatin1String( "true" ), Qt::CaseInsensitive ) == 0 )
+  {
+    layoutItem->setDrawAnnotations( true );
+  }
+  else
+  {
+    layoutItem->setDrawAnnotations( false );
+  }
+
+  layoutItem->mLayerStyleOverrides.clear();
+
+  //mLayers
+  layoutItem->mLayers.clear();
+
+  QDomNodeList layerSetNodeList = itemElem.elementsByTagName( QStringLiteral( "LayerSet" ) );
+  if ( !layerSetNodeList.isEmpty() )
+  {
+    QDomElement layerSetElem = layerSetNodeList.at( 0 ).toElement();
+    QDomNodeList layerIdNodeList = layerSetElem.elementsByTagName( QStringLiteral( "Layer" ) );
+    layoutItem->mLayers.reserve( layerIdNodeList.size() );
+    for ( int i = 0; i < layerIdNodeList.size(); ++i )
+    {
+      QDomElement layerElem = layerIdNodeList.at( i ).toElement();
+      QString layerId = layerElem.text();
+      QString layerName = layerElem.attribute( QStringLiteral( "name" ) );
+      QString layerSource = layerElem.attribute( QStringLiteral( "source" ) );
+      QString layerProvider = layerElem.attribute( QStringLiteral( "provider" ) );
+
+      QgsMapLayerRef ref( layerId, layerName, layerSource, layerProvider );
+      ref.resolveWeakly( project );
+      layoutItem->mLayers << ref;
+    }
+  }
+
+  // override styles
+  QDomNodeList layerStylesNodeList = itemElem.elementsByTagName( QStringLiteral( "LayerStyles" ) );
+  layoutItem->mKeepLayerStyles = !layerStylesNodeList.isEmpty();
+  if ( layoutItem->mKeepLayerStyles )
+  {
+    QDomElement layerStylesElem = layerStylesNodeList.at( 0 ).toElement();
+    QDomNodeList layerStyleNodeList = layerStylesElem.elementsByTagName( QStringLiteral( "LayerStyle" ) );
+    for ( int i = 0; i < layerStyleNodeList.size(); ++i )
+    {
+      const QDomElement &layerStyleElement = layerStyleNodeList.at( i ).toElement();
+      QString layerId = layerStyleElement.attribute( QStringLiteral( "layerid" ) );
+      QString layerName = layerStyleElement.attribute( QStringLiteral( "name" ) );
+      QString layerSource = layerStyleElement.attribute( QStringLiteral( "source" ) );
+      QString layerProvider = layerStyleElement.attribute( QStringLiteral( "provider" ) );
+      QgsMapLayerRef ref( layerId, layerName, layerSource, layerProvider );
+      ref.resolveWeakly( project );
+
+      QgsMapLayerStyle style;
+      style.readXml( layerStyleElement );
+      layoutItem->mLayerStyleOverrides.insert( ref.layerId, style.xmlData() );
+    }
+  }
+
+  layoutItem->mDrawing = false;
+  layoutItem->mNumCachedLayers = 0;
+  layoutItem->mCacheInvalidated = true;
+
+  //overviews
+  layoutItem->mOverviewStack->readXml( itemElem, doc, context );
```

Comment:
just confirming - did you double check that the grid/overview readXml methods work ok with composition xml? (from memory they haven't changed, but I don't trust my memory :P)

Reply:
There is a test with an overview and it apparently works, I didn't explicitly test the grid though (unless it's in the main map test).